inference-engine/src/gna_plugin/layers/gna_layer_info.hpp

```diff
@@ -90,6 +90,30 @@ class LayerInfo {
         static InferenceEngine::details::caseless_set<std::string> layersWithConstrains = {"memory", "convolution"};
         return layersWithConstrains.find(name) != layersWithConstrains.end();
     }
+    size_t getOutputBatchSize() const {
+        if (!layer) {
+            THROW_GNA_EXCEPTION << "layer is null";
+        }
+        if (!layer->outData[0]) {
+            THROW_GNA_EXCEPTION << "output data of layer '" << layer->name << "' is null";
+        }
+        auto& dims = layer->outData[0]->getDims();
+        auto layout = layer->outData[0]->getLayout();
+        switch (dims.size()) {
+        case 2:
+            if (layout == InferenceEngine::Layout::NC) {
+                return dims[0];
+            } else if (layout == InferenceEngine::Layout::CN) {
+                return dims[1];
+            } else {
+                THROW_GNA_EXCEPTION << "batch size is not define";
+            }
+        case 4:
+            return dims[0];
+        default:
+            THROW_GNA_EXCEPTION << "batch size is not define";
```

Comment:
defined
please, add layer name to this exception, otherwise it's difficult to debug

Reply:
done